src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadPropertiesStep.java

```diff
@@ -25,74 +25,31 @@
 package org.jenkinsci.plugins.pipeline.utility.steps.conf;
 
 import hudson.Extension;
-import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
-import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
-import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.pipeline.utility.steps.AbstractFileOrTextStep;
+import org.jenkinsci.plugins.pipeline.utility.steps.AbstractFileOrTextStepDescriptorImpl;
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.DataBoundSetter;
 
 import java.util.Map;
 
-import static org.apache.commons.lang.StringUtils.isBlank;
-
 /**
  * Reads java properties formatted files and texts into a map.
  *
  * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
  */
-public class ReadPropertiesStep extends AbstractStepImpl {
-    private String file;
-    private String text;
-    private Map<String, Object> defaults;
+public class ReadPropertiesStep extends AbstractFileOrTextStep {
+    private Map defaults;
```

Comment:
OK, have you filed a bug report about that?

Reply:
@rsandell, I looked into this a bit more and it looks like it has been fixed upstream a long while ago, but the dependencies declared in this plugin use very old versions of workflow-step-api. As of version 2.0 of workflow-step-api, `newInstance()` now uses the structs plugin's `DescribableModel` class instead, which appears to have a fix for this issue since early 2016.

Is there a reason the pom.xml of this plugin still points to old versions of the workflow-step-api? If not, I can open a PR to bump them.

(Also, thanks for merging and releasing!)

Reply:
Pure laziness and lack of time. There is more work to do than just bump the dependencies;

* The pom needs to be adapted to plugin parent 2.x
* then bump the dependencies at the same time, inclusing core to something like 2.7.3 at least
* All the `@Inject` and `@StepContextParameters` et.al. needs to be changed and adapted to the new way of doing things.

Most of those things are not strictly needed, but I prefer this plugin to be as "pure" as possible and that desire made it just a bit too overwhelming for me to find time to do it.
If you're up for it you are welcome to.